openedx/core/djangoapps/programs/tests/test_utils.py

```diff
@@ -116,7 +123,17 @@ def test_no_programs(self, mock_get_programs):
         self._assert_progress(meter)
         self.assertEqual(meter.completed_programs, [])
 
-    def test_single_program_engagement(self, mock_get_programs):
+    def test_entitlements_but_no_programs(self, mock_get_programs):
+        """ Verify engaged_programs is empty when entitlements exist, but no matching programs do. """
+        mock_get_programs.return_value = []
+
+        course_uuid = uuid.uuid4()
```

Comment:
Nit: If you are not using this uuid why not just rely on the CourseEntitlementFactory to assign one?

This is not a required change, just an open question.

Reply:
The method _create_entitlements will only actually create them if uuids are provided, but I don't need the variable to be separately defined since it isn't used anywhere else in the test, so I can update that.